secure_message/repository/modifier.py

```diff
@@ -97,3 +99,45 @@ def mark_message_as_read(message, user):
                 raise InternalServerError(description="Error adding read information to message")
         Modifier.remove_label(unread, message, user)
         return True
+
+    @staticmethod
+    def close_conversation(metadata, user):
+        bound_logger = logger.bind(conversation_id=metadata.id, user_id=user.user_uuid)
+        bound_logger.info("Getting user details")
+
+        user_details = internal_user_service.get_user_details(user.user_uuid)
+        bound_logger.info("Successfully retrieved user details")
+
+        try:
+            bound_logger.info("Closing conversation")
+            metadata.is_closed = True
+            metadata.closed_at = datetime.utcnow()
+            metadata.closed_by = f"{user_details.get('firstName')} {user_details.get('lastName')}"
+            metadata.closed_by_uuid = user.user_uuid
+            db.session.commit()
+        except SQLAlchemyError:
+            db.session.rollback()
+            bound_logger.exception("Database error occured while closing conversation")
+            raise InternalServerError(description="Database error occured while closing conversation")
+
+        bound_logger.info("Successfully closed conversation")
+        bound_logger.unbind('conversation_id', 'user_id')
+
+    @staticmethod
+    def open_conversation(metadata, user):
+        bound_logger = logger.bind(conversation_id=metadata.id, user_id=user.user_uuid)
+
+        try:
+            bound_logger.info("Re-opening conversation")
+            metadata.is_closed = False
+            metadata.closed_at = None
+            metadata.closed_by = ''
+            metadata.closed_by_uuid = ''
```

Comment:
`closed_by` and `closed_by_uuid` should be set to `None` to match what is done when a `Conversation` class is instantiated